api/session/auth_session.py

```diff
@@ -35,6 +38,58 @@ def user():
                 error_response(404, "Session token expired or invalid")
         error_response(404, "No session token found")
 
+    @classmethod
+    def clear_session(cls):
+        """
+        Clears the user session (signing them out)
+        Also deletes an existing jwt auth cookie and removes the corresponding
+        link records from redis
+
+        Returns: 302 redirect to signed-out page
+        """
+        existing_auth_token = request.cookies.get(
+            Config.FSD_USER_TOKEN_COOKIE_NAME
+        )
+        status = "no_token"
+        valid_token = False
+        if existing_auth_token:
+            # Check if token is valid
+            try:
+                valid_token = validate_token(existing_auth_token)
+                status = "sign_out_request"
+            except jwt.PyJWTError:
+                status = "invalid_token"
+
+            if not valid_token:
+                # Check if token has just expired
+                try:
+                    valid_token = decode_with_options(
+                        existing_auth_token, options={"verify_exp": False}
+                    )
+                except jwt.PyJWTError:
+                    status = "expired_token"
```

Comment:
Nitpicky - took me a while to understand this so I googled, could you not just catch a `jwt.ExpiredSignatureError` in the first try block and set `expired_token` there? https://pyjwt.readthedocs.io/en/latest/usage.html#expiration-time-claim-exp

Reply:
Refactored, thanks Sarah!